feat(iota-framework): replace storage_fund with storage deposits

Dkwcs opened the pull request:
# Description of change

Replacing storage_fund with storage deposits
Increase storage_rebate_rate to 100%
Removing rebates from the total_object_storage_rebates
Putting refundable charges into total_object_storage_rebates
Putting non-refundable charges (which are gonna be 0 for now if we set the refund rate to 100%) into non_refundable_balance
## Links to any relevant issues

fixes #939 and #948 .

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation Fix

## How the change has been tested

Run tests - kinesis/crates/iota-framework/packages/iota-system

Dkwcs committed:
feat(iota-framework): replace storage_fund with storage deposits

PhilippGackstatter reviewed (commented):
Looks good. You linked issue #938 but that issue is not fixed by this PR.

Dkwcs committed:
refactor(iota-procol-configs): change storage_rebate_rate to 100%, minor docs fixes

Dkwcs committed:
refactor(iota-framework): split storage_balance into 2 fields like was in storage_fund.
Removing rebates from the total_object_storage_rebates
Putting refundable charges into total_object_storage_rebates
Putting non-refundable charges (which are gonna be 0 for now if we set the refund rate to 100%) into non_refundable_balance

PhilippGackstatter reviewed (commented):
There are hundreds of results in the codebase when searching for the string `"storage_fund"`. Do you want to update all of them in this PR, or do it later in a pure-rename only PR? I think the latter is preferable, to keep the logic changes separate from the name changes, but we need to keep track of this issue somewhere.

Dkwcs committed:
minor  review fixes

Dkwcs committed:
Merge branch 'scl1/rid-storage-fund's-ability-earn-staking-rewards' into scl1/replace-storage-fund-with-storage-deposits

Dkwcs committed:
Merge branch 'scl1/rid-storage-fund's-ability-earn-staking-rewards' into scl1/replace-storage-fund-with-storage-deposits

Dkwcs committed:
Merge branch 'scl1/rid-storage-fund's-ability-earn-staking-rewards' into scl1/replace-storage-fund-with-storage-deposits

Dkwcs committed:
feat(iota-cost, iota-protocol-config): updating snapshots

Dkwcs committed:
Merge branch 'scl1/rid-storage-fund's-ability-earn-staking-rewards' into scl1/replace-storage-fund-with-storage-deposits
